src/shell.c

```diff
@@ -165,19 +165,40 @@ int main() {
 
         int argc = parse(input, command, args);
 
-        pid_t pid = vfork();
-
-        if (pid < 0) {
-            perror("fork");
-            exit(-1);
-        } else if (pid == 0) {
-            exec(command, args, argc);
-            exit(0);
-        } else {
-            wait(NULL);
+        if (strcmp(command, "cd") == 0) {
+            if (argc != 2) 
+                printf("Couldn't change directory\n");
+
+            else {
+                if (chdir(args[1]) == -1) 
+                    printf("Error from chdir\n");
+                        
+                else {
+                    char cwd[1024];
+                    if (getcwd(cwd, sizeof(cwd)) == NULL) 
+                        printf("Error from getcwd\n");
+                    else 
+                        printf("Current directory: %s\n", cwd);
+                }
+            }
         }
-    }
 
+        else {
```

Comment:
Don't use an else here, it makes the code harder to read. Put a "continue"  in the if statement instead.